File: WoodworkingProjectManager/src/WoodworkingProjectManager.Core/Model/ToolAggregate/Tool.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace WoodworkingProjectManager.Core;

public class Tool
{
    public Guid ToolId { get; set; }
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Brand { get; set; }
    public string? Model { get; set; }
    public string? Description { get; set; }
    public decimal? PurchasePrice { get; set; }
    public DateTime? PurchaseDate { get; set; }
    public string? Location { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: WoodworkingProjectManager/src/WoodworkingProjectManager.WebApp/projects/woodworking-project-manager/src/app/models/tool.ts
export interface Tool {
  toolId: string;
  userId: string;
  name: string;
  brand?: string;
  model?: string;
  description?: string;
  purchasePrice?: number;
  purchaseDate?: string;
  location?: string;
  notes?: string;
  createdAt: string;
}
